diff --git a/src/etc/get-snapshot.py b/src/etc/get-snapshot.py
index 886a84b..ed2a7bf 100755
--- a/src/etc/get-snapshot.py
+++ b/src/etc/get-snapshot.py
@@ -54,6 +54,10 @@ if len(sys.argv) == 3:
 else:
   snap = determine_curr_snapshot(triple)
   dl = os.path.join(download_dir_base, snap)
+
+  if triple == "arm-unknown-linux-gnueabihf":
+      download_url_base = "http://ftp.floorchan.org/mirror/stages/rust"
+
   url = download_url_base + "/" + snap
   print("determined most recent snapshot: " + snap)
 
diff --git a/src/snapshots.txt b/src/snapshots.txt
index b779b56..55b4a5d 100644
--- a/src/snapshots.txt
+++ b/src/snapshots.txt
@@ -1,4 +1,5 @@
 S 2014-09-22 437179e
+  linux-arm a0c71eea6f052d22071610906fafbb99e5ee6d51
   freebsd-x86_64 f693c0441de3dbb2d471dde5a5045ac8a48807d8
   linux-i386 5c2132b65f45c21b43d28de6a9460978b1a7b08a
   linux-x86_64 152be582853c2cf1ddcc88b085153b52ebbeb065
